### ISSUE TITLE
brave-dark-mode-block flag no longer works because fingerprinting protections can no longer be strict

### ISSUE THREAD

**Issue by bughit** [brave/brave-browser#38181]
## Description 

brave-dark-mode-block provided more than fingerprinting protection. I don't want web content defaulting to dark just because the android theme is dark. Its an invalid assumption that the two must always go together. So brave-dark-mode-block provided control over this. When flipped on along with strict fingerprinting protection, web content would default to light while the android theme was dark.

Now that strict fingerprinting is gone, this flag should apply to the fingerprinting protection being on.

Additionally, I would suggest turning it into a normal setting that's independent of fingerprinting protection. Because its entirely possible to prefer light content regardless of fingerprinting.

## Steps to reproduce 

   1. enable dark theme in android
   3. enable chrome://flags/#brave-dark-mode-block and fingerprinting protection
   4. load github.com or google.com or reddit.com

## Actual result 

web content defaults to dark

## Expected result

it should continue to be possible to default web content to light when the android theme is dark

## Issue reproduces how often 

100%

## Version/Channel Information:

- Can you reproduce this issue with the current Play Store version?
  - yes
- Can you reproduce this issue with the current Play Store Beta version?
  - have not tried 
- Can you reproduce this issue with the current Play Store Nightly version?
  - have not tried

## Device details
   - Install type (ARM, x86): arm
   - Device type (Phone, Tablet, Phablet): phone
   - Android version: 14
   
## Brave version

1.65.126

**Comment by TEMP-ad** [comment]
Strict fingerprinting is NOT gone, it is a flag and it is disabled by default because it was useless to 99% of people and broke too many websites that needed WebGL.

Strict Fingerprinting didn't do much compared to Standard while breaking tons of pages, this is why Brave team already said in their blogpost that they will add some of the few (like 2 of 3, no WebGL one) features of Strict to Standard.

Instead of making this issue, you could have just find the flag to get the strict fingerprinting and move on.

**Comment by arch-btw** [comment]
Related: https://github.com/brave/brave-browser/issues/38613

**Comment by bsclifton** [comment]
The flags are considered experimental - we haven't made any changes to this (as far as I know)

That said - thanks for reporting. I wasn't aware of this flag
cc: @ShivanKaul

**Comment by ghost** [comment]
`brave://flags/#brave-dark-mode-block` was actually made, so people who didn't want the strict fingerprinting mode to always give 'light' mode could turn it off and get the color-scheme if available, it wasn't to turn it on, because Strict had it.

But I don't get the issue, the flag `brave://flags/#brave-show-strict-fingerprinting-mode` still exists and has existed since Strict was replaced by the new Fingerprinting JSON exceptions, so anyone can still use Strict Fingerprinting mode if they want to.
I guess Brave has just not ported some of the Strict fingerprinting to the new single default mode. In fact, new exceptions file works perfectly fine with Strict, but strict broke too many sites and caused too many issues to be viable, so WebGL will keep breaking and etc etc, but it's strange how people are talking about a flag that was meant to be turned off to get Theme detection, but somehow the one to enable strict fingerprinting support again is ignored.

**Comment by ShivanKaul** [comment]
Right, if you want to use this flag, you should be able to enable both `brave://flags/#brave-show-strict-fingerprinting-mode` and `brave://flags/#brave-dark-mode-block` to get back the functionality. The latter flag's description says: `Always report light mode when fingerprinting protections set to Strict`. We could modify the feature flag to control always reporting light mode independent of fingerprinting protections. 

We will have to think about how to handle the case where a user has already turned on `brave://flags/#brave-show-strict-fingerprinting-mode`.

**Comment by jfrech** [comment]
> `brave://flags/#brave-dark-mode-block` was actually made, so people who didn't want the strict fingerprinting mode to always give 'light' mode could turn it off and get the color-scheme if available, it wasn't to turn it on, because Strict had it.

I think there has been a degradation: 

    DISABLED brave://flags/#brave-dark-mode-block
    ENABLED  brave://flags/#brave-show-strict-fingerprinting-mode
    private browsing
    Brave 1.85.118 Chromium: 143.0.7499.169 (Official Build) (64-bit)

*does* block dark mode, whereas non-private browsing does not. In my recollection, this behaviour is only exhibited the last few weeks.

Although `Sec-CH-Prefers-Color-Scheme` (.cf.web https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Sec-CH-Prefers-Color-Scheme 2025-12-19) is a high-entropy hint, I would argue having Brave respond to the media query induces better privacy than the realistic alternative of logging in to the Web service to change its settings.

With the synthesized dark mode `brave://flags/#enable-force-dark`, I didn't have much luck: it garbles some HI elements.
